### ISSUE TITLE
differentiate between custom blocks having the same name

### ISSUE THREAD

**Issue by pdawyndt** [scratch4d/scratch-LN#59]
```
define move to (direction)  @id1

define move to (direction)  @id2

move to {90} ::use(id2) 
```

**Comment by Ellen102** [comment]
The grammar must be changed to allow round brackets without escaping. So I change:
- the DoubleColon token to modifier 
- only a id, comment, new modifier can follow so a simplified version of the label pattern: `/::((:(?!:))|[^\{\|\\#@: \t\n]|\\[^])([ \t]*((:(?!:))|[^\|\\#@: \t]|\\[^]))*/`
 - the modifiers rule

**Comment by Ellen102** [comment]
Question: 
What happens in this case?
```
define move to (direction)  @id1

define move to (direction)  @id2

move to {90}
```
First occurrence or last occurrence?

**Comment by pdawyndt** [comment]
@Ellen102 I have no preference. Only important to be consistent, e.g. all choices should be either first or last definition. Also make sure this feature is documented.

**Comment by Ellen102** [comment]
I think that this will not be allowed anymore in future versions of Scratch. I see methods to verify a procedure name is used only once and is it not working correctly in the current version.

**Comment by pdawyndt** [comment]
@Ellen102 any idea about shadowing built-in blocks with custom blocks having the same name; will this be allowed?

**Comment by Ellen102** [comment]
As far as I understand it is not a problem, I do not see any checks for this. It is also not confusing: for the user, it is a different coloured block and in the code the type is different. I will add it such that a custom modifier can be added like `move {10} steps ::custom`

**Comment by Ellen102** [comment]
Wontfix
